Makefile

```diff
@@ -27,6 +27,13 @@ BIN_PATH = $(CURDIR)/bin
 GOROOT ?= $(DEPS_PATH)/go
 CARGO_TARGET_DIR ?= $(CURDIR)/target
 
+BUILD_INFO_GIT_FALLBACK := "Unknown (no git or not git repo)"
+BUILD_INFO_RUSTC_FALLBACK := "Unknown"
+export TIKV_BUILD_TIME := $(shell date -u '+%Y-%m-%d %I:%M:%S')
+export TIKV_BUILD_GIT_HASH := $(shell git rev-parse HEAD 2> /dev/null || echo ${BUILD_INFO_GIT_FALLBACK})
```

Comment:
Will tikv-server include the latest hash if I commit the local changes?

Reply:
If there is no source change after you committed local change, then the hash is not changed. Notice that this behavior is the same for current master:

```
# breezewish @ ~/Work/PingCAP/tikv on git:master x [13:45:16] C:130
$ target/debug/tikv-server --version
TiKV 
Release Version:   2.1.0-beta
Git Commit Hash:   8f1124735e591d255bde6a24bb4907aab1aa9c41
Git Commit Branch: master
UTC Build Time:    2018-07-31 05:43:34
Rust Version:      1.29.0-nightly (4f3c7a472 2018-07-17)

# breezewish @ ~/Work/PingCAP/tikv on git:master x [13:45:24] 
$ git commit --allow-empty -m "test"
[master c2c27d584] test

# breezewish @ ~/Work/PingCAP/tikv on git:master x [13:45:30] 
$ make dev
info: component 'rustfmt-preview' for target 'x86_64-apple-darwin' is up to date
info: component 'clippy-preview' for target 'x86_64-apple-darwin' is up to date
    Finished dev [unoptimized + debuginfo] target(s) in 0.89s                                                                                                                                                    
# When SIP is enabled, DYLD_LIBRARY_PATH will not work in subshell, so we have to set it
# again here. LOCAL_DIR is defined in .travis.yml.
export DYLD_LIBRARY_PATH=":/lib" && \
	export LOG_LEVEL=DEBUG && \
	export RUST_BACKTRACE=1 && \
	cargo test --features "default sse"  -- --nocapture && \
	cargo test --features "default sse" --bench benches  -- --nocapture  && \
	if [[ "`uname`" == "Linux" ]]; then \
		export MALLOC_CONF=prof:true,prof_active:false && \
		cargo test --features "default sse"  --bin tikv-server -- --nocapture --ignored; \
	fi
    Finished dev [unoptimized + debuginfo] target(s) in 2.49s

# breezewish @ ~/Work/PingCAP/tikv on git:master x [13:45:38] C:130
$ target/debug/tikv-server --version
TiKV 
Release Version:   2.1.0-beta
Git Commit Hash:   8f1124735e591d255bde6a24bb4907aab1aa9c41
Git Commit Branch: master
UTC Build Time:    2018-07-31 05:43:34
Rust Version:      1.29.0-nightly (4f3c7a472 2018-07-17)
```